### DISCUSSION TITLE
Sentinel 2 Data Requests

### DISCUSSION THREAD

**Discussion by clkruse** [microsoft/PlanetaryComputer#5] [Data request]
Hello,

I work with a group called [Earthrise](https://earthrise.media/), and I focus on developing models and algorithms for various earth-observation tasks. A core project that we're working on is to detect aggregations of plastic waste. We've trained sets of neural networks to find and monitor plastic waste sites, and have run the system on Southeast Asia.

Our existing networks have been trained on Sentinel 2 L1C (top of atmosphere) data. We prefer L1C over L2A as the catalog stretches back further (mid-2016 for L1C vs. early 2018 for L2A). We may be able to use L2A for future projects, but we'd need to re-generate all training data and retrain networks if we wanted to run the plastics pipeline on the planetary computer. 

Bruno had mentioned that this dataset may be in the pipeline, but I wanted to raise a use case that goes along with it.

L1C data is most important, but it would also be awesome to get the [S2Cloudless ](https://github.com/sentinel-hub/sentinel2-cloud-detector) dataset as well. Sentinel cloud filtering is passable, but not great. Better cloud masks always help for our work.

Caleb

**Comment by agentmorris** [comment]
Thanks for your interest!

**Re: L1C data...**

Our Sentinel-2 L2A archive goes back to the beginning of the Sentinel-2 mission in 2015.  That doesn't solve your problem for existing models, but for future work, you shouldn't have to worry about temporal coverage of the Sentinel-2 L2A data.

We do have an undocumented rolling window of recent L1C data (the last 90 days), in the original SAFE/jp2 format provided by ESA.  This is probably also not that useful for your scenario, but email us at planetarycomputer@microsoft.com if you need access to this.

**Re: cloud masks...**

We are working on improved cloud masks; these should be ready in a couple months.  Keep an eye on our <a href="https://planetarycomputer.microsoft.com/docs/overview/changelog/">changelog</a>!

Thanks.

-Dan

**Comment by clkruse** [comment]
Thanks for the info, Dan! I now see that the S2 coverage is the full extent of the catalog. I think I assumed that older data wasn't there because I combined my experience using other platforms and also saw that the explorer gui only goes back to 2018. This removes a concern about using the data for future projects.

And I'm excited by new cloud masks. I'll keep an eye out!

**Comment by WalternativE** [comment]
Just wanted to jump into the discussion and add my appreciation for all of the work you have been doing to increase the coverage of Sentinel-2 data products in the Planetary Computer data catalog. 🙏

If it is still time to add items to the "wish list" would it be possible to add the cloud mask and the pixel based cloud probability to the L2A products in a similar way as it is done in [Sentinel Hub](https://docs.sentinel-hub.com/api/latest/data/sentinel-2-l2a/)? The possibility of having `CLP` and `CLM` bands would be extremely convenient.

**Reply by TomAugspurger** [reply]
We're always taking wish list items :)

We're hoping to have those same cloud masks / cloud probability bands based on s2cloudless "soon". Don't have an exact timeline, yet unfortunately, but sometime in the next 3-6 months should be safe.

**Reply by WalternativE** [reply]
Thank you very much for the update! Looking forward to seeing the new catalog items 😊
Talking about wish lists...are you planning to add ERA5 Reanalyse data as well? [AWS](https://registry.opendata.aws/ecmwf-era5/) currently has a ZARR version of the dataset in their catalogs which is a massive improvement to the Copernicus Platform (nothing against Copernicus...GRIB2 is just cumbersome to work with). 😅

**Reply by TomAugspurger** [reply]
Thanks. I opened https://github.com/microsoft/PlanetaryComputer/discussions/21 for the ERA5 reanalysis.

**Reply by szwiep** [reply]
Hi @TomAugspurger, I was wondering if there are any further updates regarding the timeline for cloud masks / probability bands based on s2cloudless? Thanks!

**Reply by maxaragon** [reply]
Any updates on s2cloudless masks?

**Comment by jgomezdans** [comment]
Another plug for the Level 1 Sentinel 2 product here. For quite a lot of things, uncertainty is required, and it would be beneficial to trace the uncertainty via the atmospheric correction, which is missing in the L2A product.

**Comment by Kongstad** [comment]
Any changes in this regarding Sentinel-2 Cloud masks?

**Comment by TomAugspurger** [comment]
No updates. We'll post here when we have anything to share.

________________________________

**Reply by J6767** [reply]
Any update Re. S2 cloudless on PC? Rough ETA? Many thanks.

**Comment by TomAugspurger** [comment]
No updates here, but we'll post when we do.
________________________________
